packages/react/src/header/Header.tsx

```diff
@@ -1,36 +1,146 @@
+  isSearch: true;
+  /** Placeholder text for the search field. */
+  placeholder?: string;
+};
+
+type WithoutSearchProps = {
+  /** Determines if the search field is displayed. */
+  isSearch?: false;
+  placeholder?: never;
+  loadOptions?: never;
+  onChange?: never;
+  searchFieldVariant?: never;
+};
+
+type AsyncSearchProps<T> = WithSearchProps & {
+  /** "async" provides a dropdown list with results. */
+  searchFieldVariant: "async";
+  /** Only for "async" search field: Function to fetch/select options based on input. */
+  loadOptions: (inputValue: string, callback: (options: OptionsOrGroups<T, GroupBase<T>>) => void) => void;
+  /** 2. Async: Callback for when the selection changes. */
+  onChange: (newValue: T | null) => void;
+};
+
+type RegularSearchProps = WithSearchProps & {
+  /** Determines which variant of searchfield to use. "regular" provides a simple input field. */
+  searchFieldVariant?: "regular";
+  loadOptions?: never;
+  /** 1. Regular: Callback for when the input changes. */
+  onChange?: ChangeEventHandler<HTMLInputElement>;
 };
 
-export const Header = <T extends unknown>({ isSearch, loadOptions, onChange }: HeaderProps<T>) => {
+export type HeaderProps<T> = BaseHeaderProps & (AsyncSearchProps<T> | RegularSearchProps | WithoutSearchProps);
+
+export const Header = <T extends unknown>(props: HeaderProps<T>) => {
+  const {
+    isSearch = false,
+    loadOptions,
+    onChange,
+    justifyContent = "space-between",
+    placeholder = "Søk her...",
+    searchFieldVariant = "regular",
+    onLogoClick,
+  } = props;
+
+  const [isSm] = useMediaQuery("(max-width: 30em)");
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+
+  const logoSize = isSm ? 110 : 150;
+  const headerSize = isSm ? 70 : 90;
+  const showLogo = !isSearching || !isSm;
+  const showSearchField = isSearch && (!isSm || isSearching);
+  const showSearchButton = isSearch && !isSearching && isSm;
+  const showExitButton = isSearching && isSm;
+  const justify = justifyContent && isSm ? "space-between" : justifyContent;
+
   return (
     <Flex
-      bg={"white"}
-      borderBottomWidth={"2px"}
-      borderBottomColor={"gray.200"}
-      py={"29px"}
-      paddingLeft={"70px"}
-      alignItems={"center"}
+      bg="white"
+      borderBottomWidth="2px"
+      borderBottomColor="gray.200"
+      padding={30}
+      height={headerSize}
+      alignItems="center"
+      justifyContent={justify}
+      gap={90}
     >
-      <Box>
-        <Logo variant={"horizontal"} size={170} />
-      </Box>
-
-      {loadOptions !== undefined && onChange !== undefined && isSearch ? (
-        <Box px={"90px"} w={700} paddingTop="10px">
-          <SearchAsync
-            loadOptions={loadOptions}
-            onChange={onChange}
-            size={"lg"}
-            dropdownIndicator={<Icon icon="search" weight={400} />}
-          />
+      {showLogo && (
+        <Box onClick={onLogoClick} cursor={onLogoClick && "pointer"}>
```

Comment:
Her ville jeg nok heller ha lagt logoen inni en lenke, og så sendt med href som prop i stedet for onClick. Da standardiserer vi action på logoclick til å ta deg til en ny side (som enten bør være kartverket.no eller forsiden til applikasjonen du står i). Det tror jeg er det brukerne forventer at skal skje når de klikker på logoen. Kan evt også vurdere å legge inn kartverket.no som default for href-proppen.

Reply:
Det var lurt tenkt😄